engine/runtime-with-polyglot/src/test/java/org/enso/interpreter/test/VerifyArrowTest.java

```diff
@@ -0,0 +1,155 @@
+package org.enso.interpreter.test;
+
+import static org.junit.Assert.*;
+
+import com.oracle.truffle.api.interop.InteropLibrary;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.temporal.Temporal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.*;
+import org.enso.polyglot.RuntimeOptions;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Value;
+import org.graalvm.polyglot.io.IOAccess;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class VerifyArrowTest {
+  private static Context ctx;
+  private static Handler handler;
+
+  private static final InteropLibrary interop = InteropLibrary.getUncached();
+
+  @BeforeClass
+  public static void initEnsoContext() {
+    handler = new MockHandler();
+    ctx =
+        Context.newBuilder()
+            .allowExperimentalOptions(true)
+            .allowIO(IOAccess.ALL)
+            .option(
+                RuntimeOptions.LANGUAGE_HOME_OVERRIDE,
+                Paths.get("../../distribution/component").toFile().getAbsolutePath())
+            .option(RuntimeOptions.LOG_LEVEL, Level.FINEST.getName())
+            .logHandler(handler)
+            .out(System.out)
+            .err(System.err)
+            .allowAllAccess(true)
+            .build();
+    assertNotNull("Enso language is supported", ctx.getEngine().getLanguages().get("enso"));
+    var fourtyTwo =
+        ctx.eval("enso", "mul x y = x * y").invokeMember("eval_expression", "mul").execute(6, 7);
+    assertEquals(42, fourtyTwo.asInt());
+  }
+
+  @AfterClass
+  public static void closeEnsoContext() throws Exception {
+    if (ctx != null) {
+      ctx.close();
+    }
+  }
+
+  @Test
+  public void arrowDate32() {
+    var arrow = ctx.getEngine().getLanguages().get("arrow");
+    assertNotNull("Arrow is available", arrow);
+    var date32Constr = ctx.eval("arrow", "new[Date32]");
+
+    Value date32Array = date32Constr.newInstance((long) 10);
```

Comment:
Is the cast to `long` currently needed? Ideally we want to accept any Java number.